Manage.c

```diff
@@ -766,15 +764,15 @@ mdadm_status_t manage_add_external(struct supertype *st, int fd, char *disk_name
 	rv = MDADM_STATUS_SUCCESS;
 
 out:
-	close(container_fd);
+	close_fd(&container_fd);
 	dev_policy_free(pols);
 
 	if (sra)
 		sysfs_free(sra);
 
 	if (rv != MDADM_STATUS_SUCCESS && is_fd_valid(disk_fd))
 		/* Metadata handler records this descriptor, so release it only on failure. */
-		close(disk_fd);
+		close_fd(&disk_fd);
```

Comment:
if you changed close then `is_fd_valid()` from if criteria can be removed.

Reply:
fixed.